store/postgres/src/deployment_store.rs

```diff
@@ -288,9 +299,11 @@ impl DeploymentStore {
                 .filter(|type_name| type_name != &key.entity_type),
         );
 
+        let entity_ids: Vec<_> = keys.map(|key| &*key.entity_id).collect();
+
         if !types_with_shared_interface.is_empty() {
             if let Some(conflicting_entity) =
```

Comment:
In theory, we should be careful not to surpass PostgreSQL binding limit (which is ~64k bindings per query).
Since this query would spend only 2 bindings per conflicting ObjectType (one for the `id` array and the other for the table name), it's unlikely that this error will happen. 

If we wanted do be extra careful, we could always iterate in chunks of ~32,000 pair of values.